docs/getting-started/first-app.md

```diff
@@ -139,35 +139,32 @@ For the purpose of this tutorial, we'll now add another two roles: “Admin” a
 ![](./assets/pic-5.jpg)
 
 :::info
-
-You can also set special permissions for specific fields. For example, you can prevent users from editing the Due Date of a project and allowing it only to managers.
-
-To do so, make sure to use `Granular` permissions, and click on**`+ Add Field`** below the role list.
-
-This option is not available for the **Delete** action
+You can set action permissions for specific fields, instead of globally for all the action's fields. For example, you can prevent users from editing the Due Date of a project, while permitting managers to do so.
+To specify permissions at the field level:
+1. Set the action's permission scope to "Granular".
+2. Click **+ Add Field**. (For the *Delete* action there's no option to set permissions for specific fields.) 
+3. Select the field(s) from the list.  
+4. The roles you've defined for your app in general are listed beneath the action. To have them available for the fields, you'll need to select them at the action top level. So either select all roles, or only the roles you'll be using for these fields. 
+5. For each field set which role(s) can perform the action. For example, only Admins can create project owners, if this role was assigned to the Project entity's *Create* action's *owner* field.  
 :::
```

Comment:
It is working a little bit differently than what is described. see this:

1. Set the action's permission scope to "Granular".
2. The roles you've defined for your app in general are listed beneath the action. Select which roles can perform the action on the entire entity.
3. To limit specific fields, click **+ Add Field**. (For the *Delete* action there's no option to set permissions for specific fields.) 
4. Select the field from the list.  
5. Set which role(s) can perform the action on the specific field. 
For example, if "Manager' is assigned to the Project entity's *Create* action's *owner* field - only Managers can set the project owners when creating new Projects, other roles can create Projects but cannot set the Project owner on creation.

Reply:
What you did in the new step 3 is to repeat the information that they can limit permissions of specific fields. So you created a redundancy. 
What you  missed is that what I wrote in "my" step 4 is how it works, unfortunately--since it's very awkward.
"The roles you've defined for your app in general are listed beneath the action. To have them available for the fields, you'll need to select them at the action top level. So either select all roles, or only the roles you'll be using for these fields. "
Try it for yourself and see.

Reply:
> To have them available for the fields,

I have a problem with this part "To have them available for the fields,"... the user should select the roles on the action level - not to have them available for the fields but primarily to set which role can perform the action. 
selecting the roles at the action level is required when selecting "granular" even without selecting any field

Reply:
@udanna ?

Reply:
[https://drive.google.com/file/d/1L-fw_MmxTKsoexhry4YdTBxAEkpj1EUU/view?usp=sharing](url)

Reply:
In the above clip tried to explain what I see.

Reply:
I saw the clip. Thanks. 
Everything you mentioned there is true but the reason behind it is where we still have some differences.

The use IS NOT selecting the roles at the action level in order to have them available at the field level.
The user IS selecting the roles at the action level to determine which roles can perform the action. 

After that being said - when selecting roles at the field level, only the ones that were selected at the action level are available as you showed in the clip.

Reply:
@yuval-hazaz  I can't edit my patch.5, so what's the best flow at this point?

Reply:
@yuval-hazaz clicked on convert to draft. (How did you have this option, while I don't?) I'll see if it works.